### PR TITLE
PLANNER-2000 Replace Travis CI with Github Actions

### PR DESCRIPTION
**Pull request by triceo**

### CODE REVIEW

**Comment by triceo** [comment]
In order to enable Sonar properly, we need to add `SONARCLOUD_TOKEN` among secrets in Github. I don't seem to have the permissions to be able to do that. Christopher, can you please take care of it?

**Comment by Christopher-Chianelli** [comment]
@triceo it seems I don't have permission to do that either; can't access repository's settings. We probably need the KIE group maintainer to set it for us.

**Comment by triceo** [comment]
@Christopher-Chianelli Can you please take care of that? Although I'm really not sure why Sonar's even done in Travis.

**Comment by yurloc** [comment]
Because it's faster.

**Comment by Christopher-Chianelli** [comment]
@triceo Done; "SONARCLOUD_TOKEN" is defined; thanks @rsynek!

**Comment by triceo** [comment]
@rsynek graciously agreed to look into the Sonarcloud failure.

File: pom.xml [hunk 1]
```diff
@@ -198,54 +198,5 @@
...
-        <property>
-          <name>env.TRAVIS_EVENT_TYPE</name>
-          <value>pull_request</value>
-        </property>
-      </activation>
-      <!-- See https://docs.travis-ci.com/user/environment-variables/ for list of Travis environment variables. -->
-      <!-- See https://docs.sonarqube.org/latest/analysis/pull-request/ for properties required by Sonar for PR analysis. -->
-      <properties>
-        <!--suppress UnresolvedMavenProperty -->
-        <sonar.pullrequest.branch>${env.TRAVIS_PULL_REQUEST_BRANCH}</sonar.pullrequest.branch>
```
Comment (rsynek) [inline pom.xml:228]: How does GitHub Actions CI provide these properties? Without these, the PR won't get paired with SonarCloud.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -2,6 +2,9 @@ name: CI
 
 on: [push, pull_request]
 
+env:
+  SONAR_TOKEN: ${{ secrets.SONARCLOUD_TOKEN }}
```
Comment (rsynek) [inline .github/workflows/ci.yml:6]: Is it required to use `SONAR_TOKEN` environment variable name as the `SONARCLOUD_TOKEN` has been already taken by the secret? If so, we can rename the secret (see the next comment).

File: pom.xml (hunk 1 above)
Comment (triceo) [reply pom.xml:225]: Somehow Sonar can read them from the environment, see the log - it has all the info.

**Comment by triceo** [comment]
Ok, together with @rsynek we've finally discovered why Sonar doesn't work: As a security measure, the `SONARCLOUD_TOKEN` secret is not passed to PRs from external repositories.

Travis had the same problem, but there we actually circumvented the security precautions and set the secret to env variable directly in Travis settings. Github Actions doesn't allow that, and good for them.

The only solution here is to rely on Jenkins to run Sonar analysis for us. And since most of the KIE repositories can live with that, I would argue that we can, too. 

I'm merging this PR as is - if there are any other solutions to the problem, I'll be happy to discuss them in a follow-up conversation.

**Comment by triceo** [comment]
The following PR enables Sonarcloud for this repo in Jenkins:
https://github.com/kiegroup/kie-jenkins-scripts/pull/751

**Pull request merged**